### ISSUE TITLE
Cannot specify externalAuth true and a proxy user.

### ISSUE THREAD

**Issue by APop123** [oracle/node-oracledb#1628]
1. What versions are you using?
```
>     process.platform
'win32'
>     process.version
'v18.12.1'
>     process.arch
'x64'
>     require('oracledb').versionString
'6.2.0'
>     require('oracledb').oracleClientVersionString
undefined
```
Server version unrelated - error ocurs before connection.

2. Describe the problem

Following example 4 from https://blogs.oracle.com/opal/post/external-and-proxy-connection-syntax-examples-for-node-oracledb.

With externalAuth true and proxy user set results in the error:

```Error: NJS-136: user name and password cannot be set when using external authentication```

Is there any way to use external auth and proxy user with current version of the library?

4. Include a runnable Node.js script that shows the problem.
```
const oracledb = require('oracledb');

async function run() {
    oracledb.initOracleClient();

    await oracledb.getConnection({
        connectString: 'db.example.com:1521/myservice',
        externalAuth: true,
        user: '[myproxyuser]'
    });
}

run()
```

**Comment by cjbj** [comment]
Are you sure you're using Thick mode?  NJS-136 is shown in Thin mode with your testcase when I remove the `initOracleClient()` call.  This particular message seems like a buglet to me - I would have expected `NJS-089: External Authentication is not supported by node-oracledb in Thin mode`

**Comment by APop123** [comment]
Hi, yes definitely in thick mode (oracledb.thin is false). The issue is caused by this bit of _verifyOptions:

https://github.com/oracle/node-oracledb/blob/f4ce93077e793091887fa999a4597a31e6ac7502/lib/oracledb.js#L499-L502

Nothing here checks for the valid case of a proxy user and will error whenever user is set.

If I comment out this validation my code will connect to the DB and I can see my proxy user is set by checking ```sys_context('USERENV', 'PROXY_USER')```.

**Comment by cjbj** [comment]
Sounds like a regression to me.

**Comment by sharadraju** [comment]
The patch for this issue has been added [here](https://github.com/oracle/node-oracledb/commit/6c8ecf5c4a6718d1fc629dea6d0042fc816bac22). It will be fixed in the 6.3 release.

**Comment by sharadraju** [comment]
This has been fixed as part of the 6.3 release.
